receiver/mongodbatlasreceiver/factory.go

```diff
@@ -50,24 +52,49 @@ func createMetricsReceiver(
 	consumer consumer.Metrics,
 ) (component.MetricsReceiver, error) {
 	cfg := rConf.(*Config)
-	ms, err := newMongoDBAtlasScraper(params, cfg)
+	recv, err := newMongoDBAtlasReceiver(params, cfg)
 	if err != nil {
 		return nil, fmt.Errorf("unable to create a MongoDB Atlas Receiver instance: %w", err)
 	}
 
+	ms, err := newMongoDBAtlasScraper(recv)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create a MongoDB Atlas Scaper instance: %w", err)
+	}
+
 	return scraperhelper.NewScraperControllerReceiver(&cfg.ScraperControllerSettings, params, consumer, scraperhelper.AddScraper(ms))
 }
 
-func createLogsReceiver(
+func createCombinedLogReceiver(
 	_ context.Context,
 	params component.ReceiverCreateSettings,
 	rConf config.Receiver,
 	consumer consumer.Logs,
 ) (component.LogsReceiver, error) {
 	cfg := rConf.(*Config)
-	recv, err := newAlertsReceiver(params.Logger, cfg.Alerts, consumer)
-	if err != nil {
-		return nil, fmt.Errorf("unable to create a MongoDB Atlas Receiver instance: %w", err)
+
+	var err error
+	recv := &combindedLogsReceiver{}
+
+	if !cfg.Alerts.Enabled && !cfg.Logs.Enabled {
+		return nil, errors.New("one of 'alerts' or 'logs' must be enabled")
+	}
+
+	// If alerts is enabled create alerts receiver
+	if cfg.Alerts.Enabled {
+		recv.alerts, err = newAlertsReceiver(params.Logger, cfg.Alerts, consumer)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create a MongoDB Atlas Alerts Receiver instance: %w", err)
+		}
+	}
+
+	// If logs is enabled create logs receiver
+	if cfg.Logs.Enabled {
+		recv.logs, err = newMongoDBAtlasLogsReceiver(params, cfg)
+		if err != nil {
+			return nil, fmt.Errorf("unable to create a MongoDB Atlas Logs Receiver instance: %w", err)
+		}
+		recv.logs.consumer = consumer
```

Comment:
Why not pass `consumer` into the constructor like the alerts func? Also, you. might as well pass the logger like the alerts func too.